spec/models/info_request_spec.rb

```diff
@@ -1742,6 +1742,63 @@
     end
   end
 
+  describe '.guess_by_incoming_subject' do
+    subject { described_class.guess_by_incoming_subject(subject_line) }
+    let(:info_request) { FactoryBot.create(:info_request) }
+
+    context 'a direct reply to the original request email' do
+      let(:subject_line) { info_request.email_subject_followup }
+
+      let(:guess) do
+        described_class::Guess.new(info_request, subject_line, :subject)
+      end
+
+      it { is_expected.to include(guess) }
+
+      context 'the incoming subject has different capitalisation' do
+        let(:subject_line) do
+          info_request.email_subject_followup.gsub('Re: ', 'RE: ')
+        end
+
+        it { is_expected.to include(guess) }
+      end
+
+    end
+
+    context 'a direct reply to an original request email which matches multiple requests' do
+      let!(:info_request_1) do
+        FactoryBot.create(:info_request, title: 'How many jelly beans?')
+      end
+
+      let!(:info_request_2) do
+        FactoryBot.create(:info_request, title: 'How many jelly beans?')
+      end
+
+      let(:subject_line) do
+        'Re: Freedom of Information request - How many jelly beans?'
+      end
+
+      let(:guess_1) do
+        described_class::Guess.new(info_request_1, subject_line, :subject)
+      end
+
+      let(:guess_2) do
+        described_class::Guess.new(info_request_2, subject_line, :subject)
+      end
+
+      it { is_expected.to match_array([guess_1, guess_2]) }
+    end
+
```

Comment:
This is tricky - unless (highly likely) I'm missing something, it's not reasonable to try to do a reverse lookup and see if any of our 250k request titles look like any given subject line.

From a short sample of 200 recent replies, we seem to get the following general types:

Replies using the reply button in an email client (or something that looks like that):
"Re: [FOI boilerplate] - [Request title]"
Action: yay, we can probably catch most of these by comparing with the title and the subject format!
ToDo: what if there's a batch?

Replies - in some cases from a case management system - using the authority's own reference number:
"Information request - FOI/ABC/11/18/12345", ""Request for information - Ref: 1234567"
Action: look hopefully for previous replies using (ending with?) the same wording (with `Re`, `FW` and so on)

Replies using own ref to multiple requests (uh oh)
"FOI refs 1234 & 1235"

Too general to be useful, potentially loads of matches:
"Auto-reply", "Auto-response", "Thank you for your email", "RE: FOIA Acknowledgement", "your freedom of information request", "Information Request Response", "We have received your email", "An update on your request" etc

Noted special cases:
"[FOI-1234] Freedom of Information request - Council assets (revised)"
"Automatic reply: Freedom of Information request - [looks like the title]"
"FW: FW: Freedom of Information request - [looks like the title]"

Observations:
* Occasionally the whole subject line is in upper case or lower so both class as realistic tests.
* Using a hyphen as a subject separator seems common so that's a reasonable guess/assumption (for the UK anyway)

Reply:
Trying to avoid language-specific assumptions such as "about your request" etc which we're not going to have translations for